File: mobile/buka_app/lib/services/zodiac_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'storage_service.dart';
import 'api_config.dart';

class ZodiacService {
  static Future<Map<String, dynamic>> getZodiac(int month, int day) async {
    final baseUrl = await ApiConfig.getZodiacBaseUrl();
    final token = await StorageService.getToken();

    final response = await http.get(
      Uri.parse("$baseUrl/$month/$day"),
      headers: {
        "Content-Type": "application/json",
        "Authorization": "Bearer $token",
      },
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception("Failed to load zodiac data");
    }
  }

  static Future<List<dynamic>> getHistory() async {
    final baseUrl = await ApiConfig.getZodiacBaseUrl();
    final token = await StorageService.getToken();
    
    final response = await http.get(
      Uri.parse(baseUrl + "/history"),
      headers: {
        "Authorization": "Bearer $token",
      },
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception("Failed to load history");
    }
  }
}

File: web/buka-web/src/services/zodiacService.ts
import api from './api';
import type { ZodiacResult, HistoryItem } from '../types';
import toast from 'react-hot-toast';

export const zodiacService = {
  async getZodiac(month: number, day: number): Promise<ZodiacResult> {
    try {
      const response = await api.get(`/zodiac/${month}/${day}`);
      toast.success(`You are a ${response.data.sign}!`);
      return response.data;
    } catch (error: any) {
      const message = error.response?.data?.message || 'Failed to get zodiac';
      toast.error(message);
      throw new Error(message);
    }
  },

  async getHistory(): Promise<HistoryItem[]> {
    try {
      const response = await api.get('/zodiac/history');
      return response.data;
    } catch (error: any) {
      const message = error.response?.data?.message || 'Failed to load history';
      toast.error(message);
      throw new Error(message);
    }
  },
};
